File: cudoviste/cudoviste.py
spaces = [0, 0, 0, 0, 0]
h, w = [int(x)-1 for x in input().split(" ")]
L1 = input()
for i in range(h):
    L2 = input()
    for j in range(w):
        spots = L1[j:j+2] + L2[j:j+2]
        if "#" in spots:
            continue
        spaces[spots.count("X")] += 1
    L1 = L2
for space in spaces:
    print(space)

File: cudoviste/cudoviste.java
import java.util.Scanner;

public class cudoviste {
    public static void main(String[] args) {
        int[] spaces = new int[5];
        Scanner scn = new Scanner(System.in);
        int h = scn.nextInt();
        int w = scn.nextInt();
        scn.nextLine();
        String l1 = scn.nextLine();
        for (int i = 0; i< h-1; i++) {
            String l2 = scn.nextLine();
            for (int j = 0; j<w-1; j++) {
                String spots = l1.substring(j, j+2) + l2.substring(j, j+2);
                boolean hasBuilding = false;
                int count = 0;
                for (char c : spots.toCharArray()) {
                    if (c == '#') {
                        hasBuilding = true;
                    }
                    if (c == 'X') {
                        count++;
                    }
                }
                if (!hasBuilding) {
                    spaces[count]++;
                } 
            }
            l1 = l2;
        }
        scn.close();
        for (int space : spaces) {
            System.out.println(space);
        }
    }
}